sockeye/inference.py

```diff
@@ -67,18 +68,22 @@ def __init__(self,
                  max_output_length_num_stds: int = C.DEFAULT_NUM_STD_MAX_OUTPUT_LENGTH,
                  decoder_return_logit_inputs: bool = False,
                  cache_output_layer_w_b: bool = False,
-                 forced_max_output_len: Optional[int] = None) -> None:
+                 forced_max_output_len: Optional[int] = None,
+                 skip_softmax: bool = False) -> None:
         super().__init__(config)
         self.params_fname = params_fname
         self.context = context
         self.beam_size = beam_size
         utils.check_condition(beam_size < self.config.vocab_target_size,
                               'The beam size must be smaller than the target vocabulary size.')
+        if skip_softmax:
+            utils.check_condition(beam_size == 1, 'Skipping softmax does not have any effect for beam size > 1')
```

Comment:
as this will raise a RuntimeError, maybe: "Softmax skipping can only be enabled with beam size of 1."

Reply:
also, actually now skip_softmax is no longer a user facing parameter. So I in that case I'd say make it an assertion.